src/extensions/scratch3_text2speech/index.js

```diff
@@ -96,38 +242,76 @@ class Scratch3SpeakBlocks {
     getViewerLanguageCode () {
         // @todo This should be the language code of the project *creator*
         // rather than the project viewer.
-        return navigator.language || navigator.userLanguage || 'en-US';
+        // @todo Amazon Polly needs the locale in a two part form (e.g. ja-JP),
```

Comment:
The server could (should?) be doing this lookup. Mind creating an issue on the server side?

Reply:
all set